providers/base/api/handlers/request_handler.go

```diff
@@ -30,6 +33,9 @@ type RequestHandlerImpl struct {
 	headers map[string]string
 }
 
+// NoOpRequestHandler is a no-op RequestHandler that does nothing.
+type NoOpRequestHandler struct{}
```

Comment:
Where are we using this?

Reply:
testing

Reply:
Why can't we use mocks instead of a Nop implementation